lib/ansible/galaxy/collection/__init__.py

```diff
@@ -142,8 +147,8 @@ def __init__(self, collection_name):  # type: (str) -> None
 
 def verify_local_collection(
         local_collection, remote_collection,
-        artifacts_manager,
-):  # type: (Candidate, Optional[Candidate], ConcreteArtifactsManager) -> CollectionVerifyResult
+        artifacts_manager, keyring,
```

Comment:
Consider attaching keyring to the artifacts manager instance. I believe it's the best place architecture-wise because it already abstracts away dealing with the artifacts and so it's only right to keep the verifier close. Plus, you wouldn't need to pass the keyring object around.

Reply:
Oh, good idea.